src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -52,6 +52,8 @@ internal class GrpcWorkerChannel : IRpcWorkerChannel, IDisposable
         private readonly Dictionary<MsgType, Queue<PendingItem>> _pendingActions = new ();
         private readonly ChannelWriter<OutboundGrpcEvent> _outbound;
         private readonly ChannelReader<InboundGrpcEvent> _inbound;
+        private IDictionary<string, WorkerInvocationMetrics> _invocationMetricsPerWorkerId = new Dictionary<string, WorkerInvocationMetrics>();
```

Comment:
_Feel free to correct me if my understanding below is incorrect._

Question: There will be one instance of `GrpcWorkerChannel` for each worker instance. So each of those instances is for that specific worker instance. So, do we really need a dictionary in that case?

Reply:
See my other comment about the thread safety question. If we decide to not use a dictionary, and decided to keep int variable to store the numbers, we could use the methods from [Interlocked](https://learn.microsoft.com/en-us/dotnet/api/system.threading.interlocked?view=net-6.0) to keep the operation atomic.

Reply:
I agree - I do not think we need a dictionary here. The key is `workerId`, which as far as I can tell is read-only for the life of a `GrpcWorkerChannel`. So this dictionary would have at most 1 entry in it - so I think we can safely remove the dictionary and hold this values directly.

Reply:
Let's discuss this one offline. I did deploy this code to a function app and looks like the dictionary is getting updated properly - 

```kusto
https://wawscus.kusto.windows.net/wawsprod?query=H4sIAAAAAAAEAHWQT0vDQBDF74LfYchJD6GbKB4KCj1YCFgpJl9g3Z2mq9k%2f7sy2BPzwbgRtC%2b115vfmvTfL5BQb7%2bjF93R99Q37LUaErlk9t91itYanR9CSkY3Fm1rUdVmJsqpB3M2FmIvbg2QRwqu0CMo7lsYRFJRiXzrkkpH44b44sG2yVsbxiI2oRjVgAT6e2S6lGVLWxZzkAtJ5lkPjdl7J3z7ZbTY7S7botHE9mH8YIn6lnBH2hrdH80ZfMHtDhWaHGigphUSbNJyeo5AjIGyy%2buTc9AIfNUZ4H2GdSxvCLr%2b2ZWkDSFITEKL%2fQMV%2fvtOI5SdCJX4ALraBGa4BAAA%3d&web=0

FunctionsLogs
| where TIMESTAMP >= datetime(2022-10-12 03:00:0)
| where AppName contains "surg-net-test64"
| where Summary contains "recycle" or Summary contains "Failure rate" or Summary contains "TotalInvocations"
//or Summary contains "Sending invocation request with invocationId" or Summary contains "Received successful invocation response for invocationId"
| order by PreciseTimeStamp asc
| project Summary
| take 10


Summary
--
WorkerId: 320f5298-7802-4aad-ab2a-603a539ceae8, Failure rate: 20, TotalInvocations : 5, SuccessfulInvocations : 4, AverageInvocationLatency : 0
WorkerId: 3921ad56-794d-4e8d-afc6-b92b625b48dd, Failure rate: 80, TotalInvocations : 5, SuccessfulInvocations : 1, AverageInvocationLatency : 0
Faulty WorkerId to be Recycled: 3921ad56-794d-4e8d-afc6-b92b625b48dd, Failure rate: 80
WorkerId: 320f5298-7802-4aad-ab2a-603a539ceae8, Failure rate: 0, TotalInvocations : 3, SuccessfulInvocations : 3, AverageInvocationLatency : 0
WorkerId: 3921ad56-794d-4e8d-afc6-b92b625b48dd, Failure rate: 100, TotalInvocations : 3, SuccessfulInvocations : 0, AverageInvocationLatency : 0
Faulty WorkerId to be Recycled: 3921ad56-794d-4e8d-afc6-b92b625b48dd, Failure rate: 100
WorkerId: 320f5298-7802-4aad-ab2a-603a539ceae8, Failure rate: 0, TotalInvocations : 3, SuccessfulInvocations : 3, AverageInvocationLatency : 0
WorkerId: 3921ad56-794d-4e8d-afc6-b92b625b48dd, Failure rate: 100, TotalInvocations : 3, SuccessfulInvocations : 0, AverageInvocationLatency : 0
Faulty WorkerId to be Recycled: 3921ad56-794d-4e8d-afc6-b92b625b48dd, Failure rate: 100
WorkerId: 320f5298-7802-4aad-ab2a-603a539ceae8, Failure rate: 0, TotalInvocations : 4, SuccessfulInvocations : 4, AverageInvocationLatency : 0
```

Reply:
I think we do not need the ConcurrentDictionary. This instance of `GrpcWorkerChannel` is unique for that worker (see the `Id` property on this class). So in that case, we really do not need a dictionary.